### PR TITLE
Auto-push snapshot releases and javadocs. 

### PR DESCRIPTION
**Pull request by cgruber**
Add travis configuration, secure variables, and scripts suitable to add automatic snapshot and javadoc publishing on a successful build.

### CODE REVIEW

File: util/generate-latest-docs.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+   [ "$TRAVIS_JDK_VERSION" == "oraclejdk7" ] && \
+   [ "$TRAVIS_PULL_REQUEST" == "false" ] && \
+   [ "$TRAVIS_BRANCH" == "master" ]; then
+  echo -e "Publishing javadoc...\n"
+  
+  mvn -f build-pom.xml javadoc:aggregate
+  TARGET="$(pwd)/target"
+
+  cd $HOME
+  git clone --quiet --branch=gh-pages https://${GH_TOKEN}@github.com/google/truth gh-pages > /dev/null
```
Comment (eamonnmcmanus) [inline util/generate-latest-docs.sh]: Not sure I fully understand the environment this is running in. Is $HOME empty each time? So no gh-pages directory already there?

Comment (cgruber) [inline util/generate-latest-docs.sh]: I believe that is the case.

Comment (cgruber) [inline util/generate-latest-docs.sh]: Admittedly, there is a bit of cargo-culting going on here, but this is the same pattern used in all of our other projects that do this.

Comment (sameb) [inline util/generate-latest-docs.sh]: URL is wrong.  should be auto, not truth?  same w/ the repo slug check at the top.

File: util/generate-latest-docs.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  
+  cd gh-pages
+  git config --global user.email "travis@travis-ci.org"
+  git config --global user.name "travis-ci"
+  git rm -rf api/latest 
+  mv ${TARGET}/site/apidocs api/latest
+  git add -A -f api/latest
+  git commit -m "Lastest javadoc on successful travis build $TRAVIS_BUILD_NUMBER auto-pushed to gh-pages"
```
Comment (sameb) [inline util/generate-latest-docs.sh]: FYI I noticed this typo in my original guice script the other day -- s/Lastest/Latest

File: util/publish-snapshot-on-commit.sh [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# see https://coderwall.com/p/9b_lfq
+
+if [ "$TRAVIS_REPO_SLUG" == "google/truth" ] && \
```
Comment (sameb) [inline util/publish-snapshot-on-commit.sh]: wrong project name here too

File: util/generate-latest-docs.sh (hunk 1 above)
Comment (cgruber) [inline util/generate-latest-docs.sh]: Ugh.  Yes.  Thanks sam.  You're totally correct.  I'll fix.  It sucks that you can't test this after-success stuff conditional on a particular branch like master until you actually merge it. :(

**Comment by cgruber** [comment]
Rolling a fix for your observations.  This is what I get for doing stuff before coffee.

**Pull request merged**